src/main/java/com/google/api/codegen/ApiConfig.java

```diff
@@ -61,6 +61,14 @@ public static ApiConfig createApiConfig(Model model, ConfigProto configProto) {
     }
   }
 
+  /**
+   * Creates an ApiConfig with no content. Exposed for testing.
+   */
+  public static ApiConfig createDummyApiConfig() {
```

Comment:
The advantage of this route is that the GapicProviderFactory class can be written without worrying about exposing the snippet file names separately. If you expose them separately, it increases the complexity of GapicProviderFactory; information either has to be repeated, or shared variables need to be created.